### PR TITLE
Upgrade to java 21

### PR DESCRIPTION
**Pull request by monaullah**
Started with need to bump nva-commons (on java 17) to [version 1.39.2](https://github.com/BIBSYSDEV/nva-commons/commit/74109eb64e9d2df3e884d3096f37158171a21c68) to fix bug _NP-46823 Channel register lacks CORS headers when redirecting_

Changes:
- Upgrade to gradle version 8
- Java 21
- Update checkstyle.xml
- Upgrade junit, nva-commons, jackson and wiremock versions
- Update github workflows

### CODE REVIEW

File: config/checkstyle/checkstyle.xml [hunk 1]
```diff
@@ -17,35 +17,34 @@
  -->
 
 <module name="Checker">
```
Comment (monaullah) [inline config/checkstyle/checkstyle.xml:19]: checkstyle.xml copied from [nva-data-report-api](https://github.com/BIBSYSDEV/nva-data-report-api)

File: settings.gradle [hunk 2]
```diff
@@ -1,3 +1,2 @@
-enableFeaturePreview('VERSION_CATALOGS')
```
Comment (monaullah) [inline settings.gradle:1]: [Removed in v8.0.0-M1](https://github.com/gradle/gradle/commit/29bbb70e78e70d235053ca5e4a38b0f9bcfe4cce)

File: src/test/java/no/sikt/nva/pubchannels/handler/fetch/journal/FetchJournalByIdentifierAndYearHandlerTest.java [hunk 3]
```diff
@@ -288,7 +291,8 @@ void shouldReturnRedirectWhenChannelRegistryReturnsRedirect() throws IOException
         var response = GatewayResponse.fromOutputStream(output, HttpResponse.class);
         assertEquals(HttpURLConnection.HTTP_MOVED_PERM, response.getStatusCode());
         var expectedLocation = constructExpectedLocation(newIdentifier, year);
-        assertEquals(expectedLocation, response.getHeaders().get("Location"));
+        assertEquals(expectedLocation, response.getHeaders().get(LOCATION));
+        assertEquals(WILD_CARD, response.getHeaders().get(ACCESS_CONTROL_ALLOW_ORIGIN));
```
Comment (monaullah) [inline src/test/java/no/sikt/nva/pubchannels/handler/fetch/journal/FetchJournalByIdentifierAndYearHandlerTest.java:295]: This is the actual bug being fixed with https://github.com/BIBSYSDEV/nva-commons/commit/74109eb64e9d2df3e884d3096f37158171a21c68

File: buildSrc/build.gradle [hunk 4]
```diff
@@ -10,7 +10,7 @@ repositories {
 
 wrapper {
     distributionType = Wrapper.DistributionType.ALL
-    gradleVersion = "7.2"
+    gradleVersion = "8.0"
```
Comment (brinxmat) [inline buildSrc/build.gradle:13]: 8.7 ?

File: buildSrc/src/main/groovy/nva.publication.channels.java.rootplugin.gradle [hunk 5]
```diff
@@ -1,10 +1,27 @@
...
 }
+
+reporting {
+    reports {
+        testCodeCoverageReport(JacocoCoverageReport) {
+            testType = TestSuiteType.UNIT_TEST
+        }
+        integrationTestCodeCoverageReport(JacocoCoverageReport) {
+            testType = TestSuiteType.INTEGRATION_TEST
+        }
```
Comment (brinxmat) [inline buildSrc/src/main/groovy/nva.publication.channels.java.rootplugin.gradle:20]: I wonder if this has an effect?

File: buildspec.yml [hunk 6]
```diff
@@ -4,21 +4,19 @@ phases:
...
+      # - pip3 install --upgrade awscli
       # Install aws-sam-cli
       - pip3 install -U aws-sam-cli
 
   build:
     commands:
       # Use AWS SAM to build and package the application by using AWS CloudFormation
       - sam build --debug
-#      - gradle build
+      # gradle build
```
Comment (brinxmat) [inline buildspec.yml:19]: Does this comment make sense?

File: gradle/libs.versions.toml [hunk 7]
```diff
@@ -1,15 +1,15 @@
...
+jackson = { strictly = '2.16.1' }
 mockito = { strictly = '4.5.1' }
 hamcrest = { strictly = '2.2' }
 zalando = { strictly = '0.27.1' }
 awsLambdaJavaCore = { strictly = '1.2.1' }
 log4j = { strictly = '2.17.1' }
 slf4j = { strictly = '1.8.0-beta4' }
 awslog4j2 = { strictly = '1.5.1' }
-wiremock = { strictly = '2.33.2' }
+wiremock = { strictly = '3.0.1' }
```
Comment (brinxmat) [inline gradle/libs.versions.toml:12]: The current version of wiremock is 3.6.0…we should probably fix this in commons

File: gradle/wrapper/gradle-wrapper.properties [hunk 8]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.2-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.5-bin.zip
```
Comment (brinxmat) [inline gradle/wrapper/gradle-wrapper.properties:3]: This is at odds with the wrapper version defined above

**Review by brinxmat — CHANGES_REQUESTED** [review]
Nice! I like this PR, a few comments.

File: buildSrc/src/main/groovy/nva.publication.channels.java.rootplugin.gradle (hunk 5 above)
Comment (torbjokv) [reply buildSrc/src/main/groovy/nva.publication.channels.java.rootplugin.gradle:20]: could maybe combine if with a test report step in GitHub Actions for nice smooth reports? https://github.com/marketplace/actions/test-reporter

**Review by brinxmat — APPROVED** [review]

**Pull request merged**